pkg/controllers/management/auth/auth_config.go

```diff
@@ -32,30 +37,60 @@ type CleanupService interface {
+// various auth providers.  Those fields are not present everywhere, nor are they defined in the CRD.  Given
+// that, the regular client will "eat" those internal-only fields, so in this case, we use
+// the unstructured client, losing some validation, but gaining the flexibility we require.
 type authConfigController struct {
-	users            v3.UserLister
-	authRefresher    providerrefresh.UserAuthRefresher
-	cleanup          CleanupService
-	authConfigClient controllers.AuthConfigClient
+	users                   v3.UserLister
+	authRefresher           providerrefresh.UserAuthRefresher
+	cleanup                 CleanupService
+	authConfigsUnstructured objectclient.GenericClient
 }
 
 func newAuthConfigController(context context.Context, mgmt *config.ManagementContext, scaledContext *config.ScaledContext) *authConfigController {
 	controller := &authConfigController{
-		users:            mgmt.Management.Users("").Controller().Lister(),
-		authRefresher:    providerrefresh.NewUserAuthRefresher(context, scaledContext),
-		cleanup:          cleanup.NewCleanupService(mgmt.Core.Secrets(""), mgmt.Wrangler.Mgmt),
-		authConfigClient: mgmt.Wrangler.Mgmt.AuthConfig(),
+		users:                   mgmt.Management.Users("").Controller().Lister(),
+		authRefresher:           providerrefresh.NewUserAuthRefresher(context, scaledContext),
+		cleanup:                 cleanup.NewCleanupService(mgmt.Core.Secrets(""), mgmt.Wrangler.Mgmt),
+		authConfigsUnstructured: scaledContext.Management.AuthConfigs("").ObjectClient().UnstructuredClient(),
 	}
 	return controller
 }
 
-func (ac *authConfigController) setCleanupAnnotation(obj *v3.AuthConfig, value string) (runtime.Object, error) {
-	if obj.Annotations == nil {
-		obj.Annotations = make(map[string]string)
+func (ac *authConfigController) setCleanupAnnotation(obj *v3.AuthConfig, value string) (*v3.AuthConfig, error) {
+	name := obj.Name
+	runtimeObj, err := ac.authConfigsUnstructured.Get(name, v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	unstructuredObj, ok := runtimeObj.(*unstructured.Unstructured)
+	if !ok {
+		return nil, fmt.Errorf("auth config %s is not an unstructured value", name)
+	}
+	annotations := unstructuredObj.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+	annotations[CleanupAnnotation] = value
+	unstructuredObj.SetAnnotations(annotations)
+	uobj, err := ac.authConfigsUnstructured.Update(name, unstructuredObj)
+	if err != nil {
+		return nil, fmt.Errorf("failed to update AuthConfig object: %w", err)
+	}
+	// We ned to return an AuthConfig, but Update deals in terms of unstructured objects.
+	// Given that, we need to convert the unstructured object to an AuthConfig.
+	// Normally, we'd like to use mapstructure.Decode, but its handling of embedded structs
+	// does not give us the desired result in this instnace, hence the use of json.
+	data, err := json.Marshal(uobj.(*unstructured.Unstructured).UnstructuredContent())
```

Comment:
Nit:
Given the general confusion on return types here, it might be a good idea to do the type assertion/ok check separately and handle that error independently to avoid a panic.